backend/open_api_spec.yml

```diff
@@ -0,0 +1,721 @@
+    links:
+      type: array
+      items:
+        type: object
+        additionalProperties: false
+        required:
+          - link_url
+          - link_type
+        properties:
+          link_name:
+            type: string
+          link_url:
+            type: string
+          link_type:
+            type: string
+            enum: [PROTOCOL, RAW_DATA, DOI, LAB_WEBSITE, OTHER, DATA_SOURCE]
+    dataset:
+      allOf:
+        - $ref: "#/components/schemas/dataset_preview"
+        - type: object
+          required:
+            - name
+            - revision
+            - collection_id
+            - collection_visibility
+            - revised_at
+            - is_primary_data
+            - X_normalization
+            - dataset_assets
+          properties:
+            name:
+              type: string
+            revision:
+              type: integer
+            collection_id:
+              $ref: "#/components/schemas/collection_uuid"
+            collection_visibility:
+              $ref: "#/components/schemas/visibility"
+            revised_at:
+              type: number
+            is_primary_data:
+              $ref: "#/components/schemas/is_primary_data"
+            X_normalization:
+              type: string
+            dataset_assets:
+              type: array
+              items:
+                $ref: "#/components/schemas/dataset_asset"
+            sex:
+              $ref: "#/components/schemas/ontology_element_array"
+            ethnicity:
+              $ref: "#/components/schemas/ontology_element_array"
+            development_stage:
+              $ref: "#/components/schemas/ontology_element_array"
+            dataset_deployments:
+              type: array
+              items:
+                type: object
+                properties:
+                  url:
+                    type: string
+            cell_type:
+              $ref: "#/components/schemas/ontology_element_array"
+            cell_count:
+              type: integer
+            X_approximate_distribution:
+              $ref: "#/components/schemas/distribution"
+            schema_version:
+              type: string
+    dataset_preview:
+      type: object
+      required:
+        - id
+        - curator_tag
+        - tissue
+        - assay
+        - disease
+        - organism
+        - tombstone
+      properties:
+        id:
+          $ref: "#/components/schemas/dataset_uuid"
+        curator_tag:
+          $ref: "#/components/schemas/curator_tag"
+        tissue:
+          $ref: "#/components/schemas/ontology_element_array"
+        assay:
+          $ref: "#/components/schemas/ontology_element_array"
+        disease:
+          $ref: "#/components/schemas/ontology_element_array"
+        organism:
+          $ref: "#/components/schemas/ontology_element_array"
+        tombstone:
+          type: boolean
+    dataset_asset:
+      type: object
+      properties:
+        id:
```

Comment:
The `id` of a `dataset_asset` is a duplicate of the `dataset_id` (called `id` above) of a dataset in the current portal API. It should be removed.

For example:

```
Dataset: Single-nucleus RNA-seq of the Adult Human Kidney (Version 1.0) uuid: 07854d9c-5375-4a9b-ac34-fa919d3c3686
asset[0] uuid - 07854d9c-5375-4a9b-ac34-fa919d3c3686
asset[1] uuid - 07854d9c-5375-4a9b-ac34-fa919d3c3686
asset[2] uuid - 07854d9c-5375-4a9b-ac34-fa919d3c3686

```

Reply:
Ah, got it, I didn't realize that. Thanks Brian!

Reply:
resolved (`96d5318c`)

Reply:
I think it would be helpful to always use a specific name for an id rather than relying on context.

Clarifying my point about duplication of uuids on the call. In the current portal API, each dataset in `datasets` defines an `id` which is the dataset id. This id is then duplicated in each asset in `dataset_assets` which includes the dataset id as `dataset_id` and the asset id as `id'.  We should avoid such duplication in this API. 


        "dataset_assets": [
          {
            "created_at": 1638224578.183704,
            "dataset_id": "524e045e-e74c-4e00-9884-d5c3bef3d862",
            "filename": "explorer_cxg",
            "filetype": "CXG",
            "id": "e8b0e0ce-41aa-4666-ae2c-c10098c98cc6",
            "s3_uri": "s3://hosted-cellxgene-prod/524e045e-e74c-4e00-9884-d5c3bef3d862.cxg/",
            "type": "REMIX",
            "updated_at": 1638224578.183711,
            "user_submitted": true
          },

        "id": "524e045e-e74c-4e00-9884-d5c3bef3d862",

---

In the current portal API, there is also a duplication of the `collection_id` between the collection level and the dataset level.